util/test/sub_test

```diff
@@ -202,6 +202,105 @@ def LauncherTimeoutArgs(seconds):
               useLauncherTimeout)
 
 
+# Execute command.
+# Its stdout+stderr go to sub_test's stdout+stderr.
+# Return its exit code.
+def runAndShowOutput(command):
+    sys.stdout.flush()
+    p = subprocess.Popen(command)
+    rc = p.wait()
+    return rc
+
+# Execute command. Rc = Return code.
+# Gather its combined stdout+stderr.
+# Return (its exit code, the gathered output).
+def runAndReturnRcOutput(command):
+    p = subprocess.Popen(command,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT)
+    output = p.communicate()[0]
+    return p.returncode, output
+
+# Execute command. EIEO = Error If Error Output.
+# Gather *separately* its stdout and stderr.
+# If stderr is non-empty or the exit code is non-0, report an error.
+# If errfatal, quit with Fatal(). Otherwise, 
+# return the return code (at least 1 if got stderr) and command's stdout.
+def runAndReturnRcOutputEIEO(command, errfatal=False):
```

Comment:
I think we could consolidate many of these functions into single functions differentiated with arguments, if we're willing to take advantage of python exception handling and refactor the call-sites a bit:

```python
def run_command(command, verbose=False, errfatal=False):
    """Execute command and return exit code

    Optionally print stdout+stderr if verbose==True
    Optionally raise an error upon non-empty stderr if errfatal==True
    """


# Note: will need to refactor call sites to try/except these calls, instead of
# getting the exit code in the return
def check_output(command, errfatal=False):
    """Run command and return output, raise CalledProcessError if exitcode != 0

    This is a pseudo-backport of subprocess.check_output for python 2.6
    Optionally raise an error upon non-empty stderr if errfatal==True
    """

# Instead of checking getting exit code as an argument, we'd catch the exception thrown, e.g.
cmd = ['ls', '-l']
try:
    output = check_output(cmd)
except subprocess.CalledProcessError:
    # exitcode was not zero - do some stuff



# I've included an example implementation of how these could be combined
# Note that we no longer need a stderr case, since we're not calling out to subprocess
def file_contents(filenames):
    """Return concatenated contents of file(s) as string

    Raise error if file does not exist"""

    if type(filenames) is not list:
        filenames = [filenames]
    contents = []
    for filename in filenames:
        if not os.path.isfile(filename):
            raise IOError(filename + ' not found')
        with open(filename, 'r') as handle:
            contents.extend(handle.readlines())
    return ''.join(contents)
```

What do others think?